src/Form/Input.js

```diff
@@ -52,17 +52,27 @@ const StyledInput = createComponent({
 
 const StyledTextArea = StyledInput.withComponent('textarea');
 
+const validateValueProp = (props, propName, componentName) => {
```

Comment:
@choochootrain 

Mostly following your example, this should validate our conditions:
If the input has a type of `number` validate the value prop for a `number` type
If the input has a different type validate value prop is a `string` type